include/proxy-wasm/context.h

```diff
@@ -0,0 +1,306 @@
+    clock_gettime(CLOCK_REALTIME, &tpe);
+    uint64_t t = tpe.tv_sec;
+    t *= 1000000000;
+    t += tpe.tv_nsec;
+    return t;
+  }
+  std::pair<uint32_t, string_view> getStatus() override {
+    unimplemented();
+    return std::make_pair(1, "unimplemented");
+  }
+  WasmResult getProperty(string_view /* path */, std::string * /* result */) override {
+    return unimplemented();
+  }
+  WasmResult setProperty(string_view /* key */, string_view /* value */) override {
+    return unimplemented();
+  }
+
+  // HTTP
+  WasmResult httpCall(string_view /* target */, const Pairs & /*request_headers */,
+                      string_view /* request_body */, const Pairs & /* request_trailers */,
+                      int /* timeout_milliseconds */, HttpCallToken * /* token_ptr */) override {
+    return unimplemented();
+  }
+
+  // gRPC
+  WasmResult grpcCall(string_view, string_view /* service_name */, string_view /* method_name */,
+                      string_view /* request */, const Pairs & /* initial_metadata */,
+                      std::chrono::milliseconds & /* timeout */,
+                      GrpcToken * /* token_ptr */) override {
+    return unimplemented();
+  }
+  WasmResult grpcStream(string_view /* grpc_service */, string_view /* service_name */,
+                        string_view /* method_name */, GrpcToken * /* token_ptr */) override {
+    return unimplemented();
+  }
+  WasmResult grpcClose(GrpcToken /* token */) override { return unimplemented(); }
+  WasmResult grpcCancel(GrpcToken /* token */) override { return unimplemented(); }
+  WasmResult grpcSend(GrpcToken /* token */, string_view /* message */,
+                      bool /* end_stream */) override { // stream only
+    return unimplemented();
+  }
+
+  // Metrics
+  WasmResult defineMetric(MetricType /* type */, string_view /* name */,
+                          uint32_t * /* metric_id_ptr */) override {
+    return unimplemented();
+  }
+  WasmResult incrementMetric(uint32_t /* metric_id */, int64_t /* offset */) override {
+    return unimplemented();
+  }
+  WasmResult recordMetric(uint32_t /* metric_id */, uint64_t /* value */) override {
+    return unimplemented();
+  }
+  WasmResult getMetric(uint32_t /* metric_id */, uint64_t * /* value_ptr */) override {
+    return unimplemented();
+  }
+
+  // Shared Data
+  WasmResult
+  getSharedData(string_view /* key */,
+                std::pair<std::string /* value */, uint32_t /* cas */> * /* data */) override {
+    return unimplemented();
+  }
+  WasmResult setSharedData(string_view /* key */, string_view /* value */,
+                           uint32_t /* cas */) override {
+    return unimplemented();
+  }
+
+  // Shared Queue
+  WasmResult registerSharedQueue(string_view /* queue_name */,
+                                 SharedQueueDequeueToken *token_ptr) override {
+    return unimplemented();
+  }
+  WasmResult resolveSharedQueue(string_view /* vm_id */, string_view /* queue_name */,
```

Comment:
Why this method hasn't been renamed to `lookupSharedQueue` ?

Reply:
Fixed